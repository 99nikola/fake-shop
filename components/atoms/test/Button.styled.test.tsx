import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Button from "../Button.styled";
import "jest-styled-components";

describe("Button.styled.tsx", () => {
    render(<Button />);
    const button = screen.getByRole("button");


    it("renders html button", () => {
        expect(button).toBeInTheDocument();
    });

    it("renders enabled button by default", () => {
        expect(button).toBeEnabled();
    });

    it("has attribute class", () => {
        expect(button.hasAttribute("class")).toBe(true);
    });

    it("is not hidden", () => {
        expect(button.hidden).toBe(false);
    });

    it("has background-color: darkblue", () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule("background-color", "darkblue");
    });

    it("has maring 0 by default", () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule("margin", "0");
    });

    it("takes arument {margin}", () => {
        const tree = renderer.create(<Button margin="0 10px"/>).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule("margin", "0 10px");
    });
});

export {};