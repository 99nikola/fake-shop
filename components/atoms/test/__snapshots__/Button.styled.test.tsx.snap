// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.styled.tsx has background-color: darkblue 1`] = `
.c0 {
  width: auto;
  height: 40px;
  padding: 0 20px;
  background-color: darkblue;
  border-radius: 6px;
  font-weight: 500;
  font-size: 18px;
  line-height: 26px;
  color: white;
  border: none;
  margin: 0;
}

.c0:active {
  border: 1px solid gray;
}

<button
  className="c0"
/>
`;

exports[`Button.styled.tsx has maring 0 by default 1`] = `
.c0 {
  width: auto;
  height: 40px;
  padding: 0 20px;
  background-color: darkblue;
  border-radius: 6px;
  font-weight: 500;
  font-size: 18px;
  line-height: 26px;
  color: white;
  border: none;
  margin: 0;
}

.c0:active {
  border: 1px solid gray;
}

<button
  className="c0"
/>
`;

exports[`Button.styled.tsx takes arument {margin} 1`] = `
.c0 {
  width: auto;
  height: 40px;
  padding: 0 20px;
  background-color: darkblue;
  border-radius: 6px;
  font-weight: 500;
  font-size: 18px;
  line-height: 26px;
  color: white;
  border: none;
  margin: 0 10px;
}

.c0:active {
  border: 1px solid gray;
}

<button
  className="c0"
/>
`;
